### PR TITLE
Batch auto create index cluster state updates

### PR DESCRIPTION
**Pull request by probakowski**
This change adds batching for cluster state updates to auto create index action. It will do a single reroute at the end of the batch which speeds up the operation

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/admin/indices/create/AutoCreateAction.java [hunk 1]
```diff
@@ -122,149 +142,143 @@ protected void masterOperation(
                     finalListener.onResponse(new CreateIndexResponse(false, false, indexName));
                 }
             }, finalListener::onFailure);
-            clusterService.submitStateUpdateTask(
-                "auto create [" + request.index() + "]",
-                new AckedClusterStateUpdateTask(Priority.URGENT, request, listener) {
+            AckedClusterStateUpdateTask clusterTask = new AckedClusterStateUpdateTask(Priority.URGENT, request, listener) {
```
Comment (original-brownbear) [inline server/src/main/java/org/elasticsearch/action/admin/indices/create/AutoCreateAction.java:145]: Lets add a TODO here to move this away from `AckedClusterStateUpdateTask` and use a custom task object.

**Review by original-brownbear — APPROVED** [review]
LGTM (but lets add a TODO as requested to make this nicer). For now this is a very nice speedup for our benchmarking.

**Comment by original-brownbear** [comment]
@probakowski let's only do 8.1 here IMO. No need to push this to `8.0`, it's not that much user impact to justify the backport.

File: server/src/main/java/org/elasticsearch/action/admin/indices/create/AutoCreateAction.java (hunk 1 above)
Comment (probakowski) [reply server/src/main/java/org/elasticsearch/action/admin/indices/create/AutoCreateAction.java:146]: Done

**Comment by probakowski** [comment]
Thanks @original-brownbear! That makes sense, I'll do master only

**Pull request merged**